src/Microsoft.AspNetCore.DataProtection/NullLoggerFactory.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.DataProtection
+{
+    /// <summary>
+    /// An <see cref="ILoggerFactory"/> that provides a logger that no ops.
+    /// </summary>
+    public class NullLoggerFactory : ILoggerFactory
```

Comment:
> is this used only in tests to begin with?

No. It is also used in `DataProtectionProviderFactory.GetDefaultLoggerFactory()`

> Should this move to https://github.com/aspnet/Logging/tree/dev/src/Microsoft.Extensions.Logging.Abstractions ?

Logging already has `NullLoggerFactory` in `Microsoft.Extensions.Logging.Testing`. https://github.com/aspnet/Logging/blob/dev/src/Microsoft.Extensions.Logging.Testing/NullLoggerFactory.cs. Should that be moved to `Microsoft.Extensions.Logging.Abstractions`?